### ISSUE TITLE
<outputSchemaToDefault/> generates "_" package name when generating multiple schemata

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#4168]
The schema name (in the absence of a name) should be called "defaultSchema". Likewise, the catalog name (in the absence of a name) should be called "defaultCatalog".

There are already classes that are called this way, so it would be unsurprising to have also packages of this name.

Original bug report:

> I believe I've run into a bug in the Jooq Maven plugin (jooq-codegen-maven).  It seems to me that outputSchemaToDefault flag is not interpreted correctly during code generation when it is triggered via the maven plugin. I am running 3.5.3 of the jooq maven codegen plugin as well as jooq itself.
>
> The code generation snippet in the pom file looks something like this.  
> ```xml
                   <generator>
                        <name>org.jooq.util.DefaultGenerator</name>
                        <database>
                            <name>org.jooq.util.derby.DerbyDatabase</name>
                            <includes>.*</includes>
                            <schemata>
                                <schema>
                                    <inputSchema>public</inputSchema>
                                    <outputSchemaToDefault>true</outputSchemaToDefault>
                                </schema>
                                <schema>
                                    <inputSchema>analytics</inputSchema>
                                    <outputSchema>analytics</outputSchema>
                                </schema>
                            </schemata>
                        </database>
                        <target>
                            <packageName>com.foo.model</packageName>
                            <directory>target/generated-sources/jooq</directory>
                        </target>
                    </generator>
```
> I expect code generated for the public schema to be in package com.foo.model.
Instead they end up being in com.foo.model._
It seems to be that the maven plugin interprets this the output schema to be blank and appends an underscore because it is an invalid package name.  This looks like a bug to me, unless there is a user error on my end.

----
see also:
- https://groups.google.com/forum/#!topic/jooq-user/ijCH10E1EpY
- https://github.com/jOOQ/jOOQ/issues/3960

**Comment by kaiwaldron** [comment]
Bumping this, as we are seeing the _ being used in the package name and that is now a warning in Java 8:

Warning:(4, 26) java: '_' used as an identifier
  (use of '_' as an identifier might not be supported in releases after Java SE 8)

And it looks like this is going to be a reserved identifier in Java 9. Any workarounds or fixes for this? Be nice to get rid of all of the warnings.

**Comment by lukaseder** [comment]
@kaiwaldron You can certainly work around this by implementing an appropriate generator strategy:
http://www.jooq.org/doc/latest/manual/code-generation/codegen-generatorstrategy

In order to better understand the problem, I wonder: What's your use case of using both "outputSchemaToDefault" with multiple generated schemas?

**Comment by kaiwaldron** [comment]
@lukaseder Hi Lukas, to answer your question we have one "master" schema that is named explicitly (we want qualified with the schema name), and then we have 2 other schemas which we want to be to be default (i.e., not qualified at all with schema name). Running into another issue which I'll post to the group about

**Comment by lukaseder** [comment]
@kaiwaldron Thanks for that report on the group. I've created a separate issue for this: https://github.com/jOOQ/jOOQ/issues/5593

**Comment by lukaseder** [comment]
Fixed this now. Just like there is a `DefaultSchema.java` class (there's always a schema), there will now also be a `default_schema` package

**Comment by lukaseder** [comment]
This fix introduces an incompatible change, which is why it won't be merged to 3.8. This is OK, JDK 9 will take another year to ship, so by then, people will have upgraded to jOOQ 3.9, probably
